File: isegoria/src/main/java/com/eulersbridge/isegoria/models/FriendRequest.java
package com.eulersbridge.isegoria.models;

import android.support.annotation.Nullable;

import com.squareup.moshi.Json;

public class FriendRequest {

    public long id;

    @Nullable
    public Boolean accepted;

    @Nullable
    public Boolean rejected;

    @Json(name = "requesterProfile")
    public User requester;

    @Json(name = "requestReceiverProfile")
    public User requestReceiver;
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/FriendRequest.kt
package com.eulersbridge.isegoria.network.api.models

import com.squareup.moshi.Json

class FriendRequest (
    val id: Long,
    val accepted: Boolean?,
    val rejected: Boolean?,

    @field:Json(name = "requesterProfile")
    val requester: User?,

    @field:Json(name = "requestReceiverProfile")
    val requestReceiver: User?
)
